### PR TITLE
Implement new API

### PR DESCRIPTION
**Pull request by alexdavid**
Work in progress, still have to update all the feature tests, but can you take a look at this @kevgo?

### CODE REVIEW

File: src/browser.coffee [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
     @driver = new chrome.Driver()
-
-
-  $: (selector) =>
-    new Element this, selector
+    return new Runner
+      commands:
+        call: require './commands/call'
+        getText: require './commands/get_text'
+        visit: require './commands/visit'
```
Comment (alexdavid) [inline src/browser.coffee]: TODO: need to add a command loader class to bring in all commands found in a directory

File: src/browser.coffee [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 chromeDriverPath = require('chromedriver').path
-Element = require './element'
 url = require 'url'
-webdriver = require 'selenium-webdriver'
 chrome.setDefaultService new chrome.ServiceBuilder(chromeDriverPath).build()
+Runner = require './runner'
 
 
-class Browser extends Asyncify
+class Browser
```
Comment (kevgo) [inline src/browser.coffee]: Yes!!!!!

File: src/browser.coffee [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
-    super
-    process.on 'exit', @close
+    # process.on 'exit', @close
 
     @driver = new chrome.Driver()
-
-
-  $: (selector) =>
-    new Element this, selector
+    return new Runner
```
Comment (kevgo) [inline src/browser.coffee]: `CommandQueue`?

File: src/command.coffee [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+class Command
+
+  DEFAULT_OPTIONS =
+    name: 'command'
+    timeout: 5000
+    async: yes
+
+
+  constructor: (options, @fn) ->
```
Comment (kevgo) [inline src/command.coffee]: These param names are misleading. Maybe use `arguments` here?

File: src/command.coffee [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    if @options.async
+      @_runAsync context, args...
+    else
+      @_runSync context, args...
+
+
+  _runAsync: (context, args..., done) ->
+    roca.setTimeout @options.timeout
+    @fn.call context, args..., roca (err) =>
+      if _.contains err?.message, 'exceeded timeout'
```
Comment (kevgo) [inline src/command.coffee]: Make this a Roca feature?

File: src/browser.coffee [hunk 6]
```diff
@@ -1,25 +1,25 @@
...
+      commands:
+        call: require './commands/call'
```
Comment (kevgo) [inline src/browser.coffee]:
```coffeescript
commands: [
  require './commands/call'
  ...
]
```

File: src/runner.coffee [hunk 7]
```diff
@@ -0,0 +1,33 @@
+class Runner
+
+  constructor: ({@commands, @context}) ->
+    @jobQueue = []
+    @jobQueueRunning = no
+    @wrappedCommands = {}
+
+    @wrapExtension extensionName for own extensionName of @commands
```
Comment (kevgo) [inline src/runner.coffee]: Rename to `wrapCommand`

**Comment by kevgo** [comment]
LGTM once the commands are implemented. Very cool!

File: src/browser.coffee (hunk 3 above)
Comment (alexdavid) [inline src/browser.coffee]: Done

File: src/command.coffee (hunk 4 above)
Comment (alexdavid) [inline src/command.coffee]: Done, I made options required

**Comment by kevgo** [comment]
:rocket:

**Pull request closed without merging**